pkg/services/resource.go

```diff
@@ -142,6 +148,15 @@ func (s *sqlResourceService) Update(ctx context.Context, resource *api.Resource)
 		return nil, handleUpdateError("Resource", err)
 	}
 
+	// Create the set of labels that we will add to all the resource process:
+	labels := prometheus.Labels{
+		metricsIDLabel:     updated.ID,
```

Comment:
the `id` field is the resource ID right?
if yes, such a label will lead to cardinality explosion. consider removing the label

Reply:
Good advice! we’re keeping this for two reasons: it helps to diagnose the frequent updates by a single resource, and we don’t yet have enough resources to cause cardinality issues.